intake_stac/catalog.py

```diff
@@ -125,6 +128,31 @@ def _get_metadata(self, **kwargs):
         )
 
 
+class StacItemCollection(AbstractStacCatalog):
+    """
+    Intake Catalog represeting a STAC ItemCollection
+    """
+
+    name = "stac_item_collection"
+    _stac_cls = satstac.ItemCollection
+
+    def _load(self):
+        """
+        Load the STAC Item Collection.
+        """
+        for item in self._stac_obj:
```

Comment:
@matthewhanson - I could use a sanity check on this section. 

`self._stac_obj` is a `satstac.ItemsCollection`. Is this the best way to unpack this into individual items?

Alternatively, we can/should we be grouping these items by collection?

Reply:
I'm not sure grouping is necessary. The most common use case I think is a catalog with a single collection of data. Additionally, a user might want to group them according to  any of the other metadata fields, so I think we can leave it to the user to decide how to sort/use them.